activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -1296,9 +1297,36 @@ public QueueMessageReference getMessage(String id) {
         return null;
     }
 
+    public List<MessageId> getAllMessageIds() throws Exception {
```

Comment:
here out of memory error is possible: https://github.com/apache/activemq/pull/848#discussion_r1062161446
will address it